docs/faq/tensor_inspector_tutorial.md

```diff
@@ -0,0 +1,168 @@
+
+![Screen Shot 2019-07-10 at 5 34 20 PM](https://user-images.githubusercontent.com/16669457/61013773-fe36a180-a338-11e9-9a2b-5f11ccc7afa7.png)
+
+Also, there are a bunch of built-int value checkers. Refer to the Enum below:
+
+```c++
+enum  CheckerType {
+	NegativeChecker, // check if is negative
+	PositiveChecker, // check if is positive
+	ZeroChecker, // check if is zero
+	NaNChecker, // check if is NaN, will always return false if DType is not a float type
+	InfChecker, // check if is infinity, will always return false if DType is not a float type
+	PositiveInfChecker, // check if is positive infinity,
+						// will always return false if DType is not a float type
+	NegativeInfChecker, // check if is nagative infinity,
+						// will always return false if DType is not a float type
+	FiniteChecker, // check if is finite, will always return false if DType is not a float type
+	NormalChecker, // check if is neither infinity nor NaN
+	AbnormalChecker, // chekck if is infinity or nan
+};
+```
+
+Remember the second API?
+
+```c++
+std::vector<std::vector<int>> check_value(CheckerType ct,
+		bool interactive = false, std::string tag =  "");
+```
+
+You can simply pass in a value from `CheckerType` where you would have passed in your own lambda if you were using the first API. Note that it's the developer's responsibility to pass in a valid value checker.
+
+### Dump Tensor Value
+
+Sometimes, you might want to dump the tensor to a file in binary mode. Then, you might want to use a python script to further analyze the tensor value.  Or, you might do that simply because a binary dumps has better precision and is faster to load than if you copy-paste the output from `print_string()` and load it as a `JASON` string. Either way, you can use this API:
+
+```c++
+void dump_to_file(std::string tag);
+```
+
+This API will create a file with name  "{tag}_{visit_count}.npy", where tag is the name that we give to the call, and visit is the visit count, should the operated be called more than once.
+
+The output format is `.npy`, version 1.0. This is the Numpy format and we can easily load it with the following code:
+
+```
+import numpy as np
+a = np.load('abc_1.npy')
+print(a)
+```
+
+Let's see the how it runs:
+
+![Screen Shot 2019-07-10 at 5 17 29 PM](https://user-images.githubusercontent.com/16669457/61013259-cc244000-a336-11e9-8564-a018041634f6.png)
```

Comment:
Proper host

Reply:
Yeah, I have them uploaded in another PR. Will update the url after taking in the comments